fedora/jellyfin.spec

```diff
@@ -28,7 +28,7 @@ BuildRequires:  libcurl-devel, fontconfig-devel, freetype-devel, openssl-devel,
 # COPR @dotnet-sig/dotnet or
 # https://packages.microsoft.com/rhel/7/prod/
 BuildRequires:  dotnet-runtime-3.1, dotnet-sdk-3.1
-Requires: %{name}-server = %{version}-%{release}, %{name}-web >= 1, %{name}-web < 2
+Requires: %{name}-server = %{version}-%{release}, %{name}-web >= 10.6, %{name}-web < 10.7
```

Comment:
So this implies that you are going to break compatibility with minor version updates.  Not sure if that's what you intend or not, but consider...

Version 10.6[.0] is released with a given API/set of features.  After a while, a bug needs fixing so you release 10.6.1.  Everyone using this version of Jellyfin will get the updated jellyfin-web.  All good.

But now you want to release a new Jellyfin and Jellyfin-web with a (forward-compatible) **incremental** update in the API of jellyfin-web.  That means that the new jellyfin-web is still compatible with 10.6, so will work with older Jellyfin installations, but it has new additional features that a newer Jellyfin can use.

You don't want to limit that new jellyfin-web from being used with the older Jellyfin installations because that allows you apply bug fixes to the new jellyfin-web that both older and newer Jellyfin users can enjoy, without having to fork jellyfin-web and now have to maintain and apply bug fixes to both branches.

This is how the versioning of dynamically-linked libraries on Linux works.  They are typically versioned `x.y.z` where `x` indicates a version that is incompatible with older versions because some existing API was broken.  `y` indicates newer versions of the library, with newer APIs but maintaining all previous APIs so that older consumers can continue to use that library.  `z` indicates a bugfix update.

Many dynamic libraries never make it to version 2.y.z, since they never break the existing API.  That means that the authors only ever maintain a single branch and can continue to deliver bugfixes to all consumers of the library, no matter which `1.y.*` version of the library they are using.  This obviously makes it super easy to maintain bug fixes for all users without having to manage umpteen forks and apply bug fixes to all.

So, all of that said, my advise for you, now that you are effectively separating the jellyfin-web component from jellyfin (logically -- I know it's been physically separate for a while now) and using an API to bind them, is resist the urge to try to keep their versions in sync in some way.  Move forward with the attitude that they are separate projects each with their own independent release cycle who's versions have no bearing on one another.  Try as hard as you can to never break backward compatibility -- create new APIs where you need to change the signature of a function that can wrap old APIs if you wish or have existing APIs call into the newer API with default values for new arguments, etc.

This will provide you with the least amount of support effort of jellyfin-web releases without requiring users to upgrade their jellyfin instances entirely to get bug fixes for jellyfin-web.

Reply:
Despite the packaging split, we are planning to keep the releases 100% synchronized for the foreseeable future. The packaging changes for now are mostly for build sanity rather than a true separation of them into distinct projects. For instance the default install will still always get both installed with web running under Kestrel in the main server.

Reply:
I think we will break the versions being equal sometime in the future... but definitely not until 11.X hits and we do breaking API changes. So we'll revise this when the time comes.